### ISSUE TITLE
Different share folder for media

### ISSUE THREAD

**Issue by jd1900** [sabeechen/hassio-google-drive-backup#253]
I'm looking for a different folder to "share" to store media in my Raspberry PI with Home Assistant, because when I backup that folder, it can increase way too much it size.
I'm currently using /share/media

Which folder could I use that could be shared among different docker containers? Do I need to do anything special to have a different "share" folder?

**Comment by sabeechen** [comment]
Sorry it took me a while to get back to you on this, I've been working on a bunch of bugs and this slipped off my radar.

This is more of a general question about addons, but I'll provide any advice I can.
You might already be aware, but in the settings for this addon you can choose to do a "Partial" snapshot that includes everything except the /share folder.  You can only exclude the whole /share folder (not just a subdirectory) which is a supervisor restriction.  That might solve your problem.

In terms of folder sharing between addons, whoever created the addon chooses which folders the addon can access.  The supervisor gives it 4 to choose from:
- /share, which you're familiar with
- /homeassistant - the configuration directory
- /ssl - which usually only has ssl certificates in it
- /addons - a directory you can use to test local addons

Each addon also gets its own /data folder, which is usually only accessible to the addon and nothing else.  Although each directory was created for a specific purpose there is nothing keeping you from putting any kind of file you want into any of the directories.  Most people seem to prefer to use /share for anything that doesn't fit into the "typical" categories, as you've already done.  The problem is that whatever addon you're using would need to be configured (by its developer) to access the folder you're storing things in.

I hope that answers your question, if I misunderstood let me know.

**Comment by MrYadro** [comment]
I think addon just have to support new /media folder (it's already available in new HA betas).
And user have to move everything media related from /share into /media folder. Then disable taking snapshot of it.

**Comment by jd1900** [comment]
> Sorry it took me a while to get back to you on this, I've been working on a bunch of bugs and this slipped off my radar.
> I hope that answers your question, if I misunderstood let me know.

@sabeechen , thank you for your answer. As I though there is no good solution so far, as I would like all of those folders to be backed up, as some addons as zigbee2mqtt stores it config in /share.

Therefore, the best solution would be, as @MrYadro says, to support the upcoming new /media folder. I didn't know about that, but makes a lot of sense.

**Comment by sabeechen** [comment]
I plan to add support for the /media folder in an upcoming release.  Does the supervisor support multiple media folders?

**Comment by MrYadro** [comment]
That's not multiple media folders that's /media with multiple folders inside. Really confusing term.

```
media_dirs:
    plex: /media/plex
    test_if_name_matter: /config/test
```

![image](https://user-images.githubusercontent.com/1587606/93921151-d43c8580-fd18-11ea-946f-155a75c8f6f5.png)

![image](https://user-images.githubusercontent.com/1587606/93921277-0352f700-fd19-11ea-88d4-032e28a6f366.png)

Here's supervisor backup interface:
![image](https://user-images.githubusercontent.com/1587606/93921567-617fda00-fd19-11ea-9c3b-1c31184fc664.png)

**Comment by sabeechen** [comment]
Thanks, it looks like I'll be able to support it in the same as the supervisor does.

**Comment by sabeechen** [comment]
I just released the addon version 0.102.0, which includes support for the media folder.  Please let me know or reopen this issue if you run into any trouble with the new version.
Also just a reminder, to make Home Assistant see the new version right away go to "Supervisor" > "Add-on Store" > Three dots in the top right corner > "Reload"
